lib/redis/event_handler.js

```diff
@@ -39,6 +39,25 @@ exports.connectHandler = function (self) {
 
     self.initParser();
 
+    self.lastWriteTime = null;
+    if (typeof self.options.timeoutPerCommand === 'number') {
+      var timeoutPerCommand = self.options.timeoutPerCommand;
+      debug('Per-command timeout set to %s', self.options.timeoutPerCommand);
+      self.timeoutPerCommand = setInterval(() => {
+        var commandQueueLength = self.commandQueue.length;
+        if (
+          commandQueueLength > 0 &&
+          Date.now() - self.lastWriteTime > self.options.timeoutPerCommand
```

Comment:
Since `self.lastWriteTime` is updated every time a command is added to the `commandQueue`, if the client attempts commands fast enough (more than one every `timeoutPerCommand` ms), then `self.lastWriteTime` continues to get updated such that `Date.now() - self.lastWriteTime` will continue to be less than `timeoutPerCommand` and thus the original command will never time out.

Perhaps in `Redis.prototype.sendCommand` `lastWriteTime` is only updated if `commandQueue.length === 1`.